scripts/install.sh

```diff
@@ -7,110 +7,106 @@ DONE_START='\n\e[1;32m' # new line + bold + green
+   echo -e "\t--skip-client-build: Should execute contracts part only. Client installation will not be executed.\n"
    exit 1 # Exit script after printing help
 }
 
 # Transform long options to short ones
 for arg in "$@"; do
   shift
   case "$arg" in
-    "--config-dir")     set -- "$@" "-c" ;;
-    "--network")        set -- "$@" "-n" ;;
-    "--contracts-only") set -- "$@" "-m" ;;
-    "--help")           set -- "$@" "-h" ;;
-    *)                  set -- "$@" "$arg"
+    "--network")           set -- "$@" "-n" ;;
+    "--skip-deployment")   set -- "$@" "-d" ;;
+    "--skip-client-build") set -- "$@" "-b" ;;
+    "--help")              set -- "$@" "-h" ;;
+    *)                     set -- "$@" "$arg"
   esac
 done
 
 # Parse short options
 OPTIND=1
-while getopts "c:n:mh" opt
+while getopts "n:dbh" opt
 do
    case "$opt" in
-      c ) config_dir_path="$OPTARG" ;;
       n ) network="$OPTARG" ;;
-      m ) contracts_only=true ;;
+      d ) skip_deployment=${OPTARG:-true} ;;
+      b ) skip_client_build=${OPTARG:-true} ;;
       h ) help ;;
       ? ) help ;; # Print help in case parameter is non-existent
    esac
 done
 shift $(expr $OPTIND - 1) # remove options from positional parameters
 
 # Overwrite default properties
-CONFIG_DIR_PATH=${config_dir_path:-$CONFIG_DIR_PATH_DEFAULT}
 NETWORK=${network:-$NETWORK_DEFAULT}
-CONTRACTS_ONLY=${contracts_only:-false}
+SKIP_DEPLOYMENT=${skip_deployment:-false}
+SKIP_CLIENT_BUILD=${skip_client_build:-false}
 
 # Run script
 printf "${LOG_START}Starting installation...${LOG_END}"
 
-printf "Config dir path: $CONFIG_DIR_PATH\n"
-printf "Network: $NETWORK"
+printf "Network: $NETWORK\n"
 
-cd $KEEP_CORE_SOL_PATH
+cd $KEEP_BEACON_SOL_PATH
 
-printf "${LOG_START}Installing NPM dependencies...${LOG_END}"
-npm install
+printf "${LOG_START}Installing YARN dependencies...${LOG_END}"
+# rm -rf node_modules/
+yarn install
 
-if [ "$NETWORK" == "local" ]; then
+if [ "$NETWORK" == "development" ]; then
     printf "${LOG_START}Unlocking ethereum accounts...${LOG_END}"
     KEEP_ETHEREUM_PASSWORD=$KEEP_ETHEREUM_PASSWORD \
-        npx truffle exec scripts/unlock-eth-accounts.js --network $NETWORK
+        npx hardhat unlock-accounts --network $NETWORK
 fi
 
-printf "${LOG_START}Migrating contracts...${LOG_END}"
-rm -rf build/
+printf "${LOG_START}Building contracts...${LOG_END}"
 
-CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY=$CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY \
-    npx truffle migrate --reset --network $NETWORK
+rm -rf build && rm -rf cache && rm -rf typechain
+yarn build
 
-printf "${LOG_START}Copying contract artifacts...${LOG_END}"
-rm -rf artifacts
-cp -r build/contracts artifacts
-npm link
+if [ "$SKIP_DEPLOYMENT" = false ] ; then
+   printf "${LOG_START}Deploying contracts...${LOG_END}"
+   # rm -rf deployments/development/
 
-printf "${LOG_START}Initializing contracts...${LOG_END}"
+   CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY=$CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY \
```

Comment:
How is `CONTRACT_OWNER_ETH_ACCOUNT_PRIVATE_KEY` used in the process?

Reply:
it's removed now